CFLAGS and LDFLAGS were modified for macOS in 2004.
See https://github.com/mono/libgdiplus/commit/bce17c28c445c49e9349aace2eb0e4a9bd235aad.
It is not entirely clear why the changes were needed because the bug report is no longer available.
However, they are causing problems.
See https://trac.macports.org/ticket/24501.

There is an upstream pull request to fix the issue.
See https://github.com/mono/libgdiplus/pull/534.
--- configure.orig	2012-01-31 10:28:27.000000000 -0700
+++ configure	2019-02-20 04:30:57.000000000 -0700
@@ -12527,9 +12527,6 @@
 #define OSX 1
 _ACEOF
 
-		CFLAGS="-no-cpp-precomp $CFLAGS"
-		LDFLAGS="-flat_namespace -undefined suppress $LDFLAGS"
-		no_x=yes
 		;;
 
 	*)
